client/src/modules/user/UserProfile/UserProfile.jsx

```diff
@@ -0,0 +1,236 @@
+import React, { Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { FormattedDate, FormattedMessage, FormattedTime } from 'react-intl';
+import classNames from 'classnames';
+
+import Avatar from 'common/components/Avatar';
+import { getCurrentUser } from 'modules/user/model/selectors';
+import { UserPropType } from 'common/constants/propTypes';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+import Preloader from 'common/components/Preloader';
+import { fetchUserDetails } from 'modules/user/model/actions';
+import PasswordChangeForm from 'modules/user/AuthBox/components/PasswordChangeForm';
+
+class UserProfile extends PureComponent {
+  pendingPromise = null;
+
+  state = {
+    isPasswordUpdateFormVisible: false,
+    pending: false
+  };
+
+  componentDidMount() {
+    this.handleFetchUserDetails();
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  handleFetchUserDetails = () => {
+    const {
+      currentUser: { name },
+      fetchUserDetails
+    } = this.props;
+
+    this.setState({ pending: true });
+    this.pendingPromise = makeAbortablePromise(fetchUserDetails(name));
+    this.pendingPromise.promise
+      .then(() => this.setState({ pending: false }))
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          this.setState({ pending: false });
+        }
+      });
+  };
+
+  handlePasswordChangeVisibility = event => {
+    event.preventDefault();
+    this.setState(({ isPasswordUpdateFormVisible }) => ({
+      isPasswordUpdateFormVisible: !isPasswordUpdateFormVisible
+    }));
+  };
+
+  renderHeader = () => {
```

Comment:
All these `renderX` can be extracted as separate components that are pure and easy to maintain. They would just accept simple props like `name`, `avatarUrl` etc. It is better to be done now than later.

Reply:
It's a lot of work and not a scope of this task so I created a separated task for this that can be found in JIRA under EOC_407